core/storage/suggestion/gae_models.py

```diff
@@ -1990,3 +2022,1459 @@ def export_data(
+
+            [language_code].[contributor_id]
+        """
+        return (
+            '%s.%s' % (language_code, contributor_id)
+        )
+
+    # Here we use MyPy ignore because the signature of this method
+    # doesn't match with BaseModel.get().
+    # https://mypy.readthedocs.io/en/stable/error_code_list.html#check-validity-of-overrides-override
+    @classmethod
+    def get( # type: ignore[override]
+        cls, language_code: str, contributor_id: str
+    ) -> Optional[TranslationSubmitterTotalContributionStatsModel]:
+        """Gets the TranslationSubmitterTotalContributionStatsModel
+        matching the supplied language_code and contributor_id.
+
+        Args:
+            language_code: str. ISO 639-1 language code.
+            contributor_id: str. User ID.
+
+        Returns:
+            TranslationSubmitterTotalContributionStatsModel|None. The matching
+            TranslationSubmitterTotalContributionStatsModel, or None if no
+            such model instance exists.
+        """
+        entity_id = cls.construct_id(
+            language_code, contributor_id)
+        return cls.get_by_id(entity_id)
+
+    @classmethod
+    def fetch_page(
+        cls,
+        page_size: int,
+        offset: int,
+        language_code: str,
+        sort_by: Optional[SortChoices],
+        topic_ids: Optional[List[str]],
+        num_days_since_last_activity: Optional[int]
+    ) -> Tuple[Sequence[TranslationSubmitterTotalContributionStatsModel],
+                int,
+                bool]:
+        """Returns the models according to values specified.
+
+        Args:
+            page_size: int. Number of models to fetch.
+            offset: int. Number of results to skip from the beginning of all
+                results matching the query.
+            language_code: str. The language code to get results for.
+            sort_by: SortChoices|None. A string indicating how to sort the
+                result.
+            topic_ids: List[str]|None. List of topic ID(s) to fetch
+                contributor stats.
+            num_days_since_last_activity: int. To get number of users
+                who are active in num_days_since_last_activity.
+
+        Returns:
+            3-tuple(sorted_results, next_offset, more). where:
+                sorted_results:
+                    list(TranslationSubmitterTotalContributionStatsModel).
+                    The list of models which match the supplied language_code,
+                    topic_ids and num_days_since_last_activity filters,
+                    returned in the order specified by sort_by.
+                next_offset: int. Number of results to skip in next batch.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results
+                    after this batch.
+        """
+
+        sort_options_dict = {
+            constants.CD_ADMIN_STATS_SORT_OPTIONS[
```

Comment:
I keep saying this again and again, and it doesn't seem to get addressed. To quote a previous review:

> Also, as mentioned in a previous comment, the mapping should be directly from SORT_KEY_INCREASING_LAST_ACTIVITY and you should be passing the sort keys to this function, rather than the human-readable values.

Reply:
I am confused here about how you want flow to be. If you can see on line 2287 in am passing the sort keys to this function. Then it is checking on the basis of sort_key which sort query is appropriate.

Reply:
If you're not sure about something, please feel free to ask for clarification rather than ignoring the comment.

What I am saying is that this dict should be:

```
sort_options_dict = {
     SortChoices.SORT_KEY_INCREASING_LAST_ACTIVITY.value: -cls.last_contribution_date,
     SortChoices.SORT_KEY_DECREASING_LAST_ACTIVITY.value: cls.last_contribution_date,
     ...
}
```

rather than what you have.

Reply:
So do we need to define values in constants.ts, should I remove those values?

Reply:
Thanks @seanlip PTAL at the new changes if that looks good? Fow now I have kept the changes in the constants file as well. Thinking we might require it in future while building frontend.